pkg/steps/new_e2e.go

```diff
@@ -0,0 +1,147 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	coreapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/junit"
+)
+
+type newE2ETestStep struct {
+	name             string
+	podClient        PodClient
+	jobSpec          *api.JobSpec
+	pre, tests, post []api.TestStep
+}
+
+func NewE2ETestStep(
+	config api.TestStepConfiguration,
+	podClient PodClient,
+	jobSpec *api.JobSpec,
+) (api.Step, error) {
+	return newNewE2ETestStep(config, podClient, jobSpec), nil
+}
+
+func newNewE2ETestStep(
+	config api.TestStepConfiguration,
+	podClient PodClient,
+	jobSpec *api.JobSpec,
+) *newE2ETestStep {
+	return &newE2ETestStep{
+		name:      config.As,
+		podClient: podClient,
+		jobSpec:   jobSpec,
+		pre:       config.PreSteps,
+		tests:     config.Steps,
+		post:      config.PostSteps,
+	}
+}
+
+func (s *newE2ETestStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
```

Comment:
That is added in a subsequent commit, as you probably saw after commenting, but even then it is not correct. Input resolution happens before we know which images come from `release_tag`, so it won't work.

Note that the same mechanism is used in `imageForStep`, but by then we have already imported the images. I also got strange results when I tested this where the variables were not added to `params` because the IS seemed to be empty. I'll have to investigate this.